scripts/quests/windurst/Reap_What_You_Sow.lua

```diff
@@ -0,0 +1,219 @@
+                        not (player:getFameLevel(xi.fameArea.WINDURST) >= 4 and
+                            not player:hasCompletedQuest(xi.questLog.WINDURST, xi.quest.id.windurst.LET_SLEEPING_DOGS_LIE)) -- if player fame is 4 or higher then Let Sleeping Dogs Lie takes priority if not completed
+                    then
+                        return quest:progressEvent(463, 0, xi.item.SOBBING_FUNGUS, xi.item.BAG_OF_HERB_SEEDS)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [463] = function(player, csid, option, npc)
+                    if
+                        option == 3 and
+                        npcUtil.giveItem(player, xi.item.BAG_OF_HERB_SEEDS)
+                    then
+                        quest:begin(player)
+                    end
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_ACCEPTED
+        end,
+
+        [xi.zone.WINDURST_WATERS] =
+        {
+            ['Akkeke'] = quest:event(468, 0, xi.item.POPOTO),
+
+            ['Chomoro-Kyotoro'] = quest:event(472),
+
+            ['Foi-Mui'] = quest:event(466, 0, xi.item.EAR_OF_MILLIONCORN),
+
+            ['Kirarara'] = quest:event(465),
+
+            ['Koko_Lihzeh'] = quest:event(471, 0, xi.item.FAERIE_APPLE),
+
+            ['Paku-Nakku'] = quest:event(467),
+
+            ['Mashuu-Ajuu'] =
+            {
+                onTrade = function(player, npc, trade)
+                    return handleTurnIn(trade)
+                end,
+
+                onTrigger = function(player, npc)
+                    return quest:progressEvent(464, 0, xi.item.SOBBING_FUNGUS, xi.item.BAG_OF_HERB_SEEDS)
+                end,
+            },
+
+            ['Rukuku'] = quest:event(470),
+
+            onEventFinish =
+            {
+                [475] = function(player, csid, option, npc)
+                    handleComplete(player, xi.settings.main.GIL_RATE * 500)
+                end,
+
+                [477] = function(player, csid, option, npc)
+                    handleComplete(player, xi.settings.main.GIL_RATE * 700)
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_COMPLETED and
+                quest:getMustZone(player)
+        end,
+
+        [xi.zone.WINDURST_WATERS] =
+        {
+            ['Mashuu-Ajuu'] = quest:event(476),
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_COMPLETED and
+                not quest:getMustZone(player)
+        end,
+
+        [xi.zone.WINDURST_WATERS] =
+        {
+            ['Akkeke'] = {
+                onTrigger = function(player, npc)
+                    return quest:getVar(player, 'Prog') == 1 and quest:event(468, 0, xi.item.POPOTO) or nil
```

Comment:
I'm really not a fan of the sheer number of ternaries here.  I'd suggest breaking this out into another section

Reply:
Cleaned up 👍🏻